src/app/visit/visit.cpp

```diff
@@ -42,13 +48,20 @@ void Visit::createVisit(const std::shared_ptr<Doctor>& doctor)
 {
     Visit visit(doctor);
     auto visit_ptr = std::make_shared<Visit>(visit);
-    visit.doctor_->appendVisit(visit_ptr);
-    Clinic::appendVisit(visit_ptr);
+    if (visit.doctor_.expired())
+    {
+        throw std::runtime_error("Cannot create Visit because doctor is expired!");
+    }
+    else
+    {
+        visit.doctor_.lock()->appendVisit(visit_ptr); // does it need to checkced by expired()?
```

Comment:
It should be checked at ctor. And check can be moved to separate function 
Smth like this should work:
`return doctor.expired() ? doctor : throw std::runtime_error(...);`

Reply:
Yeap, check here makes not much sense. We cant guarantee that weak_ptr will be valid seconds later. 
Any one willing to use weak_ptr<Doctor> will have to lock it anyway just prior to usage.
Besides, throwing here will make anyone creating visit to catch possible exceptions (but I don't know, maybe its not a bad idea.).